### DISCUSSION TITLE
Still does not connect

### DISCUSSION THREAD

**Discussion by yongra** [tl-its-umich-edu/my-learning-analytics#1351] [Q&A]
Hello.

I am still struggling with getting MYLA running since August. I followed the directions on both README and Wiki and yet, still it refuses to connect when I launch it from the course. I have built the server several times to see if I missed any steps. But no luck in every case. Thanks to the people who replied to my postings, I got a lot further than when I started, but still, it is not working. So, I would like to ask the following questions.

- Is it because I am setting this up on the test server?
- What ports does myla use to connect with Canvas? We have a load balancer that does translation to offload the traffic from SSL to whatever ports we need. Currently, I only have the SSL post open and it is being forwarded to 5001 on the myla server.
- Is there anyone who can help me one to one? I am OK with setting up a server, but I am not a software developer. So, I think that is why I am struggling to have it set up.

I have some more information.

- I found that the redirect URL/Target Link URI "/lti/launch/" does not work. It is as if the directory "launch" does not exist. I get 405 error.
- The JWK URL "/lti/jwks/" returns some value.
- The Open ID Inititation URL "/lti/login/" returns the following error: {"lti_error": "LTI Login failed due to missing \"target_link_uri\" param"}

Thank you.

**Comment by zqian** [comment]
@yongra did you following the [wiki page for LTI configurations](https://github.com/tl-its-umich-edu/my-learning-analytics/wiki/Deploy:-Configuration-for-LTI-1.3-Deployment)? 

Does the standalone tool work as expected? You can set the following parameter in the env.json file to test the standalone version first:

"STUDENT_DASHBOARD_LTI": false,

**Reply by yongra** [reply]
Yes, it works as a standalone with the web interface. I disabled the LTI and enabled the backend access. See the screenshot.

![image](https://user-images.githubusercontent.com/4802115/144127310-15f6cb2e-9203-4b3a-9cc1-b3a8efa60699.png)

**Comment by zqian** [comment]
@yongra this looks like the [sample data](https://drive.google.com/drive/u/0/folders/1Pj7roNjRPGyumKKal8-h5E6ukUiXTDI9) linked from the README file. What's the plan for getting Canvas data for your own Canvas instance? Eventually, MyLA will need to pull data from Canvas data output.

**Comment by yongra** [comment]
Yes. That is correct. Once I know it is working correctly, we are going to use it on the production. But, at the moment, I want to set it up on the test server first. Unless MyLA cannot be set up on the test server.


Ra

**Comment by jonespm** [comment]
MyLA uses and eventually requires data from the [Canvas Data Service](https://community.canvaslms.com/t5/Admin-Guide/What-is-Canvas-Data-Services/ta-p/142). It doesn't get any data directly from the instance. It only uses LTI for the launch.

There are 2 sources of data the Canvas Data **Dim** and **Fact** table (which are used in MyLA by Assignments, Resources and Grades) as well as Live (Caliper) Events. 
 
You need to have this data stored and loaded by the cron from a database you don't need both and if you don't have Live Events you can disable the resources view. Michigan has their Canvas Data in a warehouse provided by Unizin and I believe it's currently stored in Redshift. The Live Events we have in BigQuery. The cron is only configured to work with these and would need customizations to work in other environments. 

I believe Canvas has a paid service to host the data for some users but we're not familiar with how compatible that is with MyLA, but it should be close to the UDW format.

As far as getting it to work in test, [this says](https://community.canvaslms.com/t5/Canvas-Data-Users/Getting-Started-Canvas-Data-Services-Overview/ta-p/248030) that Canvas provides data services for their beta and production instances but we don't have a separate warehouse with separate data from the beta instance. We pull all of our data from production warehouse and do the LTI launch from test and beta instances, even though some of the data might not match 100%.
